src/overlays/actors/ovl_En_M_Fire1/z_en_m_fire1.c

```diff
@@ -18,12 +17,55 @@ const ActorInit En_M_Fire1_InitVars = {
     (ActorFunc)EnMFire1_Init,
     (ActorFunc)EnMFire1_Destroy,
     (ActorFunc)EnMFire1_Update,
-    (ActorFunc)NULL
+    NULL
 };
-*/
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_M_Fire1_0x808B5230/EnMFire1_Init.asm")
+static ColliderCylinderInit sCylinderInit = {
+    { 
+        COLTYPE_NONE, 
+        AT_ON | AT_TYPE_PLAYER, 
+        AC_NONE, 
+        OC1_NONE, 
+        OC2_TYPE_PLAYER, 
+        COLSHAPE_CYLINDER, 
+    },
+    { 
+        ELEMTYPE_UNK2, 
+        { 0x00000001, 0x00, 0x01 }, 
+        { 0xF7CFFFFF, 0x00, 0x00 }, 
+        TOUCH_ON | TOUCH_SFX_NONE, 
+        BUMP_NONE, 
+        OCELEM_NONE, 
+    },
+    { 100, 100, 0, { 0, 0, 0 } },
+};
+
+void EnMFire1_Init(Actor* thisx, GlobalContext *globalCtx) {
+    EnMFire1* this = THIS;
+    s32 pad;
+
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    if (this->actor.params != 0) {
+        this->collider.info.toucher.dmgFlags = 0x40000;
+    }
+}
+
+void EnMFire1_Destroy(Actor* thisx, GlobalContext *globalCtx) {
+    EnMFire1* this = THIS;
+    
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_M_Fire1_0x808B5230/EnMFire1_Destroy.asm")
+void EnMFire1_Update(Actor* thisx, GlobalContext *globalCtx) {
+    EnMFire1* this = THIS;
+    s32 pad;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_M_Fire1_0x808B5230/EnMFire1_Update.asm")
+    if (Math_StepToF(&this->unk_190, 1.0f, 0.2f) != 0) {
```

Comment:
Generally we like to do things like `if (Math_StepToF(&this->unk_190, 1.0f, 0.2f))` (assuming they still match of course) instead of an explicit comparison to 0